src/main/java/moze_intel/projecte/network/commands/KnowledgeCMD.java

```diff
@@ -0,0 +1,126 @@
+
+public class KnowledgeCMD {
+    private enum ActionType {
+        LEARN,
+        UNLEARN,
+        CLEAR,
+        TEST
+    }
+    public static ArgumentBuilder<CommandSourceStack, ?> register(CommandBuildContext context) {
+        return Commands.literal("knowledge")
+            .requires((source) -> source.hasPermission(2))
+            .then(Commands.literal("clear")
+                .requires(PEPermissions.COMMAND_KNOWLEDGE_CLEAR)
+                .then(Commands.argument("player", EntityArgument.player())
+                    .executes((ctx) -> handle(ctx, ActionType.CLEAR))
+                )
+            )
+            .then(Commands.literal("learn")
+                .requires(PEPermissions.COMMAND_KNOWLEDGE_LEARN)
+                .then(Commands.argument("player", EntityArgument.player())
+                    .then(Commands.argument("item", ItemArgument.item(context))
+                        .executes((ctx) -> handle(ctx, ActionType.LEARN))
+                    )
+                )
+            )
+            .then(Commands.literal("unlearn")
+                .requires(PEPermissions.COMMAND_KNOWLEDGE_UNLEARN)
+                .then(Commands.argument("player", EntityArgument.player())
+                    .then(Commands.argument("item", ItemArgument.item(context))
+                        .executes((ctx) -> handle(ctx, ActionType.UNLEARN))
+                    )
+                )
+            )
+            .then(Commands.literal("test")
+                .requires(PEPermissions.COMMAND_KNOWLEDGE_TEST)
+                .then(Commands.argument("player", EntityArgument.player())
+                    .then(Commands.argument("item", ItemArgument.item(context))
+                        .executes((ctx) -> handle(ctx, ActionType.TEST))
+                    )
+                )
+            );
+    }
+
+    private static int handle(CommandContext<CommandSourceStack> ctx, ActionType action) throws CommandSyntaxException {
+        ServerPlayer player = EntityArgument.getPlayer(ctx, "player");
+        IKnowledgeProvider provider;
+        try {
+            provider = player.getCapability(PECapabilities.KNOWLEDGE_CAPABILITY).orElseThrow(() -> new IllegalStateException("Failed to get knowledge provider."));
+        } catch (IllegalStateException ignore) {
+            ctx.getSource().sendFailure(PELang.COMMAND_PROVIDER_FAIL.translateColored(ChatFormatting.RED));
+            return 0;
+        }
+        if (action == ActionType.CLEAR) {
+            if (provider.getKnowledge().size() == 0) {
+                ctx.getSource().sendFailure(PELang.COMMAND_KNOWLEDGE_CLEAR_FAIL.translateColored(ChatFormatting.RED, player.getDisplayName()));
+                return 0;
+            }
+            provider.clearKnowledge();
+            provider.sync(player);
+            ctx.getSource().sendSuccess(PELang.COMMAND_KNOWLEDGE_CLEAR_SUCCESS.translateColored(ChatFormatting.GREEN, player.getDisplayName()), true);
+            return 1;
+        }
```

Comment:
Main differences I can see:
ProjectE:
* Succeeds regardless
* supports multiple players
This Version:
* Success is dependent on actual success of operation
* limited to one player

Reply:
Agreed, that ideally this would be modified to support multiple. Also the existing PE one should have a comment added `//TODO - 1.20: Remove, replaced be: x`

Reply:
![image](https://user-images.githubusercontent.com/17226394/194929471-382abe78-ce08-4fa3-a222-784715059e74.png)
This should be sufficient